test/widgets/autocomplete_test.dart

```diff
@@ -247,6 +308,98 @@ void main() {
+    final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+    store.setServerEmojiData(
+      ServerEmojiData(
+        codeToNames: {
+          '1f4a4': ['zzz', 'sleepy'], // Unicode emoji for "zzz"
+          '1f52a': ['biohazard'],
+          '1f92a': ['zany_face'],
+          '1f993': ['zebra'],
+          '0030-fe0f-20e3': ['zero'],
+          '1f9d0': ['zombie'],
+        },
+      ),
+    );
+    await store.handleEvent(
+      RealmEmojiUpdateEvent(
+        id: 1,
+        realmEmoji: {
+          '1': eg.realmEmojiItem(emojiCode: '1', emojiName: 'buzzing'),
+        },
+      ),
+    );
+
+    const zulipOptionLabel = 'zulip';
+    const zanyFaceOptionLabel = 'zany_face';
+    const zebraOptionLabel = 'zebra';
+    const zzzOptionLabel = 'zzz, sleepy';
+    const unicodeGlyph = '💤';
+    const zombieOptionLabel = 'zombie';
+    const zeroOptionLabel = 'zero';
+    const buzzingOptionLabel = 'buzzing';
+    const biohazardOptionLabel = 'biohazard';
+
+    // Adjust the order so the best match appears last
+    final emojiSequence = [
+      zulipOptionLabel,
+      zzzOptionLabel,
+      unicodeGlyph,
+      zanyFaceOptionLabel,
+      zebraOptionLabel,
+      zeroOptionLabel,
+      zombieOptionLabel,
+      buzzingOptionLabel,
+      // biohazardOptionLabel, this won't be rendered in the list initally since it is the 7th option.
+    ];
+
+    // Enter a query; options appear, of all three emoji types.
+    // TODO(#226): Remove this extra edit when this bug is fixed.
+    await tester.enterText(composeInputFinder, 'hi :');
+    await tester.enterText(composeInputFinder, 'hi :z');
+    await tester.pumpAndSettle();
+
+    final listViewFinder = find.byType(ListView);
+    check(because: "reason: 'ListView should be rendered'",listViewFinder,).findsOne();
+
+    final positions = emojiSequence.map((icon) {
+      final finder = find.text(icon);
+      check(because:"Each emoji option should be rendered", finder).findsOne();
+      return tester.getTopLeft(finder).dy;
+    }).toList();
+
+    for (int i = 0; i < positions.length - 1; i++) {
+      check(because: "${emojiSequence[i + 1]} should appear above ${emojiSequence[i]} because of reverse order",
+      positions[i]).isGreaterThan(positions[i + 1]);
+    }
+
+    final initialScrollOffset = tester.getTopLeft(listViewFinder).dy;
+    await tester.drag(listViewFinder, const Offset(0, -50));
+    await tester.pumpAndSettle();
+    final scrollOffsetAfterDragDown = tester.getTopLeft(listViewFinder).dy;
+
+    check(because:"ListView should not scroll down because it is already at the bottom",
+    scrollOffsetAfterDragDown).equals(initialScrollOffset);
+
+    final biohazardFinder = find.text(biohazardOptionLabel);
+    check(because: "The biohazard emoji should not be visible before scrolling up",biohazardFinder).findsNothing();
+
+    // Scroll up
+    await tester.drag(listViewFinder, const Offset(0, 50));
+    await tester.pumpAndSettle();
```

Comment:
I tried adding `print`'s before and after this, but it seems like `tester.getTopLeft(listViewFinder).dy` doesn't actually get the scroll offset, as the value also remains the same:
```dart
    print(tester.getTopLeft(listViewFinder).dy); // prints 196.0
    await tester.drag(listViewFinder, const Offset(0, 50));
    await tester.pumpAndSettle();
    print(tester.getTopLeft(listViewFinder).dy); // prints 196.0
```
This probably means that all the other checks for `scrollOffset` aren't really effective. It might be good to ask about this in #mobile-dev-help.